tests/infer/test_gradient.py

```diff
@@ -26,35 +26,35 @@ def test_subsample_gradient(Elbo, reparameterized, subsample):
     pyro.clear_param_store()
     data = torch.tensor([-0.5, 2.0])
     subsample_size = 1 if subsample else len(data)
-    num_particles = 50000
     precision = 0.06
     Normal = dist.Normal if reparameterized else fakes.NonreparameterizedNormal
 
     def model(subsample):
```

Comment:
I think you should wrap these model and guide with `@poutine.broadcast` and avoid relying on the implicit `@poutine.broadcat` due to `num_particles>1` and `vectorize_particles=True`. While it is correct without the explicit decorator, it is better style to be explicit, and users may look to our tests for idiomatic usage.

Reply:
`vectorize_particles` with `num_particles=1` would still use `@poutine.broadcast`, so as not to have some special surprising behavior for `num_particles=1`. That was part of the reason why I thought of calling it `auto_vectorize` to emphasize that its a bit more general than adding an outer iarange, and could be used a replacement for `@poutine.broadcast` with or without `num_particles`. That said, `vectorize_particles` is a clearer arg name, but it makes the broadcasting side-effect seem unexpected. Will change to using `@poutine.broadcast`, unless there is a better name that we can use for this arg.